### PR TITLE
io.Reader added

### LINKED ISSUE #1: Add Reader just like Writer
In some situations, you can't set writer. So we also need the Reader version of the current Writer.

### PR DESCRIPTION
**Pull request by gozeloglu**
* Fixed https://github.com/binalyze/ctxio/issues/1 
* tests added

### CODE REVIEW

File: reader.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"context"
+	"io"
+	"sync"
+)
+
+type Reader struct {
+	reader     io.Reader
+	ctx        context.Context
+	progressFn func(int642 int64)
```
Comment (yakuter) [inline reader.go]: This should be just progressFn func(int64).

File: reader.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	mu         sync.RWMutex
+	n          int64
+	err        error
+}
+
+func noonProgressFn(int64) {}
+
+// NewReader creates a new Reader with the given io.Reader
+func NewReader(ctx context.Context, r io.Reader, progressFn func(int642 int64)) *Reader {
```
Comment (yakuter) [inline reader.go]: Again not progressFn func(int642 int64). It should be **progressFn func(int64)**

**Review by yakuter — CHANGES_REQUESTED** [review]

**Comment by yakuter** [comment]
Thanks for the PR. There are just 2 small issues. When you change it I will merge @gozeloglu

**Comment by gozeloglu** [comment]
> Thanks for the PR. There are just 2 small issues. When you change it I will merge @gozeloglu

It was a carelessness. Sorry for that. I fixed and committed.

**Review by yakuter — APPROVED** [review]

**Comment by yakuter** [comment]
Great work thank you @gozeloglu

**Pull request merged**